packages/ui/src/machines/authenticator/actors/signIn.ts

```diff
@@ -477,7 +477,31 @@ export function signInActor({ services }: SignInMachineOptions) {
             rest = { ...rest, phone_number: phoneNumberWithCountryCode };
           }
 
-          return Auth.completeNewPassword(user, password, rest);
+          try {
+            // complete forceNewPassword flow and get updated CognitoUser
+            const newUser = await Auth.completeNewPassword(
+              user,
+              password,
+              rest
+            );
+
+            if (newUser.challengeName) {
```

Comment:
Good catch! Yeah, that should catch any issues with setup TOTP or MFA.

Reply:
e2e test caught it ;)